### ISSUE TITLE
asynk::Connection is shut down when a clone is dropped.

### ISSUE THREAD

**Issue by fd** [nats-io/nats.rs#118]
#### NATS version (`grep 'name = "nats"' Cargo.lock -A 1`)

`version = "0.8.1"`

#### rustc version (`rustc --version` - we support Rust 1.37 and up)

`rustc 1.46.0 (04488afe3 2020-08-24)`

#### Steps or code to reproduce the issue:

```rust
#[tokio::test]
async fn test_drop_clone() {
    let nats1 = nats::asynk::connect("nats://stan:4222").await.unwrap();
    let nats2 = nats1.clone();
    nats1.publish("foo", b"bar").await.unwrap();
    nats2.publish("foo", b"bar").await.unwrap();
    drop(nats1);
    nats2.publish("foo", b"bar").await.unwrap();
}
```

#### Expected result:

Connection should not shutdown when a clone is dropped.

#### Actual result:

Because the Drop handler of `nats1` is called shutdown is initiated but `nats2` is still around.

**Comment by chris13524** [comment]
I'm guessing this is a regression introduced recently by #113?

**Comment by stjepang** [comment]
This is now fixed in version 0.8.2
